enhancements/ingress/global-options-enable-hsts.md

```diff
@@ -0,0 +1,457 @@
+...
+    if ci.Spec.HTTPHeaders != nil && ci.Spec.HTTPHeaderPolicy != nil && len(ci.Spec.HSTSHeaderPolicy.MaxAge) != 0 {
+        header := "max-age=" + ci.Spec.HTTPHeaders.HSTSHeaderPolicy.MaxAgeSeconds
+        for _, directive := range ci.Spec.HTTPHeaders.HSTSHeaderPolicy.HSTSDirectives {
+            switch directive {
+                case  PreloadHSTSDirective:
+                    header = header +  ";" + PreloadHSTSDirective
+                case IncludeSubDomainsHSTSDirective:
+                    header = header + ";" + IncludeSubDomainsHSTSDirective
+            }
+        }
+        env = append(env, corev1.EnvVar{Name: RouterHSTSHeaderPolicy, Value: header})
+        switch ci.Spec.HTTPHeaders.HSTSHeaderPolicy.Scope {
+            case LimitedHSTSHeaderPolicyScope: // more details on this later in the EP doc
+                env = append(env, corev1.EnvVar{Name: RouterHSTSHeaderScope, Value: LimitedHSTSHeaderPolicyScope})
+            case AllHSTSHeaderPolicyScope:
+                env = append(env, corev1.EnvVar{Name: RouterHSTSHeaderScope, Value: AllHSTSHeaderPolicyScope})
+        }
+        }
+    }
+...
+```
+
+We don't do any other header validation, but if needed it could go in here:
+https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/operator/controller/ingress/controller.go#L441
+
+Accompanying unit and e2e test changes are added to exercise the new header.
+
+A revised HAProxy template is added which will use a new helper method called `hstsHeaderValue`, defined later.
+With the addition of processing for `ROUTER_DEFAULT_HSTS`, the code template snippet above becomes:
+```gotemplate
+{{- $router_global_hsts := env "ROUTER_DEFAULT_HSTS" "" }}
+...
+{{- if matchValues (print $cfg.TLSTermination) "edge" "reencrypt" }}
+  {{- /* Get the HSTS header value using the default and the annotations in the cfg */ }}
+  {{- with $hsts := hstsHeaderValue $router_global_hsts "" $cfg "" }}
+    http-response set-header Strict-Transport-Security {{$hsts}}
+  {{- end }}{{/* hstsHeaderValue */}}
+{{- end }}{{/* is "edge" or "reencrypt" */}}
+```
+
+#### Enable HSTS per-domain
+Some of the proposal for this goal is in the section above.  In order to allow the HAProxy template to add HSTS Headers to
+a limited list of domains, it will use the previously mentioned field `InScopeDomains` in the `IngressControllerHSTSHeaderPolicy`
+type.
+
+The details on the processing for limited scope in `desiredRouterDeployment` are completed here.
+Add one more environment variable, `ROUTER_HSTS_LIST`, which will contain the list of domains in `InScopeDomains`:
+```go
+	const RouterHSTSHeaderList        = "ROUTER_HSTS_LIST"
+
+       if scope ==  LimitedHSTSHeaderPolicyScope {
+            env = append(env, corev1.EnvVar{Name: RouterHSTSHeaderScope, Value: LimitedHSTSHeaderPolicyScope})
+            env = append(env, corev1.EnvVar{Name: RouterHSTSHeaderList, Value: ci.Spec.HTTPHeaders.HSTSHeaderPolicy.InScopeDomains})
```

Comment:
Presumably you will need something like `strings.Join(ci.Spec.HTTPHeaders.HSTSHeaderPolicy.InScopeDomains, ";")` to be able to store it in an environment variable.